protobuf/lib/src/protobuf/proto3_json.dart

```diff
@@ -89,11 +100,32 @@ Object? _writeToProto3Json(_FieldSet fs, TypeRegistry typeRegistry) {
   var result = <String, dynamic>{};
   for (var fieldInfo in fs._infosSortedByTag) {
     var value = fs._values[fieldInfo.index!];
-    if (value == null || (value is List && value.isEmpty)) {
+    var overrideForEmitsDefaults = false;
```

Comment:
I'm a bit confused about why this variable is needed. Shouldn't `context.emitDefaults` always mean that the values are always generated? I.e. `final overrideForEmitDefaults = context.emitDefaults;`, or just use `context.editDefaults` instead of a new variable.

Reply:
Another problem here is we duplicate the logic for default field values. We have some related code in `field_type.dart`. `FieldInfo` has `readonlyDefault`. We should reuse one of those instead. I think `FieldInfo.readonlyDefault` should work.

Reply:
> I'm a bit confused about why this variable is needed. Shouldn't `context.emitDefaults` always mean that the values are always generated? I.e. `final overrideForEmitDefaults = context.emitDefaults;`, or just use `context.editDefaults` instead of a new variable.

This was a good call and has been addressed.  I'll take a look at the second comment now.

Reply:
> Another problem here is we duplicate the logic for default field values. We have some related code in `field_type.dart`. `FieldInfo` has `readonlyDefault`. We should reuse one of those instead. I think `FieldInfo.readonlyDefault` should work.

Thank you for pointing me to this existing code.  This has been changed, and worked great.